### ISSUE TITLE
feat: pack and push to local repo

### ISSUE THREAD

**Issue by franroa** [GustavEikaas/easy-dotnet.nvim#89]
In my workflow I usually need to push packages locally to test them before pushing to the remote repo.

For example, given my problem:
- I have one repo with the database
- I have other many repos which connect to that database
- I want to update the database schema, and this is not an application, just a library

My solution:
- change the database schema locally, then change the Entities related to that schema 
- To tests those entities I need to add the new database version to an application
- Packing and pushing locally helps me to add the library to my local application

**Comment by franroa** [comment]
Indeed, I already have code for this, I was just wondering if this is a good improvement for the plugin
with overseer:
```
local tmpl = {
  name = "Pack Nugets",
  builder = function(params)
    return {
      name = "Pack Nugets " .. LazyVim.root.git():match("([^/]+)$"),
      cmd = "dotnet pack",
      cwd = LazyVim.root.git(),
    }
  end,
}
return tmpl

```

```
local tmpl = {
  name = "Push All Nuget Packages Locally",
  builder = function(params)
    local root_git = LazyVim.root.git()
    local command = "find " .. root_git .. " -type d -name 'Release' -exec find {} -name '*.nupkg' \\;"
    local result = vim.fn.systemlist(command)
    bash_cmd = ""
    for i, res in ipairs(result) do
      if i > 1 then
        bash_cmd = bash_cmd .. " && "
      end
      bash_cmd = bash_cmd .. "dotnet nuget push " .. res .. " --source /root/.nuget/local_source/ --api-key ''"
    end

    return {
      name = "Push nugets locally",
      cmd = bash_cmd,
      cwd = "/",
      components = { { "dependencies", task_names = { "Pack Nugets" } }, "on_exit_set_status", "default" },
    }
  end,
}
return tmpl

```

I could choose which packages to pack/push, but I think this is not needed (at least in my case, where I don't too many of them)

**Comment by GustavEikaas** [comment]
Yeah, good suggestion. Ill take a closer look once I get my head above water on the testrunner 😅

**Comment by GustavEikaas** [comment]
@franroa I know you are not working so much with this anymore but im keen to hear your input on this.

I did a similiar flow to the one you are describing today but instead of pushing the nuget package to a local repository I ended up commenting out the packagereference and adding a projectreference with the path to the local nuget package. This allows you to make changes to the package without having to pack and repush. Are there any reasons why this wont work, or the pushing to local repository will be favorable? 

One case I can imagine is if the nuget package that you are referencing are dependent on other nuget packages in the same solution. Possibly causing some duplicate package id issues

**Comment by franroa** [comment]
Hi! I am not working with .NET and I got a bit rusty, but I think your solution will not work if you want one package to be pushed to a local repo, right?. I don't this is much of a problem though

**Comment by GustavEikaas** [comment]
Correct. Im just wondering if there is any benefit to actually pushing to a local repo instead of just swapping a package reference for a project reference while testing

**Comment by franroa** [comment]
Well, if agility matters on big projects this is a concern I guess

**Comment by GustavEikaas** [comment]
Took me way too long but I made a first version of this, It can definitely be iterated on and improved in multiple PR's. But for now

1. `:Dotnet push`
2. Telescope picker for which project to pack and push
3. Build release
4. Pack release
5. Telescope picker for which source to push to, including local sources
6. dotnet push
